### ISSUE TITLE
[BUG] <<UNORDERED>> not working in nested array with in payload

### ISSUE THREAD

**Issue by ronaldfenner** [kinbiko/jsonassert#32]
### What exactly did you do?
Tried to use \<\<UNORDERED\>\> in side a payload with an array complains length doesn't match

	ja.Assertf(string(body), `
		{
			"code":200,
			"revenue":[
				"<<UNORDERED>>",
				{"date":"2020-10-01","value":100.01},
				{"date":"2020-10-02","value":200.01}
			]
		}`)

### What did you expect would happen?
It would work and validate the json

### What actually happened?
```console
    overall_structs_test.go:57: length of arrays at '$.revenue' were different. Expected array to be of length 3, but contained 2 element(s)
    overall_structs_test.go:57: actual JSON at '$.revenue' was:
        [{"date":"2020-10-01","value":100.01},{"date":"2020-10-02","value":200.01}]
        but expected JSON was:
        ["\u003c\u003cUNORDERED\u003e\u003e",{"date":"2020-10-01","value":100.01},{"date":"2020-10-02","value":200.01}]
```

### Additional info

- Output from `go version`: [e.g. `go version go1.14.5 darwin/amd64`] go version go1.17.4 darwin/amd64
- Version of this package: [e.g. `v.1.0.2`] v1.0.2

**Comment by kinbiko** [comment]
Hi @ronaldfenner, thanks for raising this.

I can't seem to reproduce this issue yet (see #33) -- please could you confirm that the *expected* payload (the payload including `<<UNORDERED>>` is written literally in the test, and not first passed through any (de)serialisation or other processing?

If that's not the source of the issue, please could you provide a runnable code example where the issue can be seen?

**Comment by kinbiko** [comment]
Oh, I figured out what happened. `<<UNORDERED>>` was only implemented in `v1.1.0-beta`, but the most "recent", according to `go mod` is `v1.0.2`... I'll cut a new release.

**Comment by kinbiko** [comment]
Please upgrade to `v1.1.0`, and your test should behave as expected. Please re-open the issue if you're still having problems.
